packages/hardhat-foundry/src/index.ts

```diff
@@ -0,0 +1,179 @@
+      );
+    }
+    return;
+  }
+
+  // Load foundry config
+  const foundryConfig = JSON.parse(runCmdSync("forge config --json"));
+
+  // Ensure required keys exist
+  if (
+    foundryConfig?.src === undefined ||
+    foundryConfig?.cache_path === undefined
+  ) {
+    throw new HardhatFoundryError(
+      "Couldn't find `src` or `cache_path` config keys after running `forge config --json`"
+    );
+  }
+
+  // Ensure foundry src path doesnt mismatch user-configured path
+  const userSourcesPath = userConfig.paths?.sources;
+  const foundrySourcesPath = foundryConfig.src;
+
+  if (userSourcesPath !== undefined && userSourcesPath !== foundrySourcesPath) {
+    throw new HardhatFoundryError(
+      `User-configured sources path (${userSourcesPath}) doesn't match path configured in foundry (${foundrySourcesPath})`
+    );
+  }
+
+  // Set sources path
+  config.paths.sources = foundryConfig.src;
+
+  // Change hardhat's cache path if it clashes with foundry's
+  if (foundryConfig.cache_path === "cache") {
+    config.paths.cache = "cache_hardhat";
+  }
+
+  pluginActivated = true;
+});
+
+// Task that translates import names to sourcenames using remappings
+internalTask(TASK_COMPILE_TRANSLATE_IMPORT_NAME).setAction(
+  async (
+    { importName }: { importName: string },
+    _hre,
+    runSuper
+  ): Promise<string> => {
+    if (!pluginActivated) {
+      return runSuper({ importName });
+    }
+
+    const remappings = getRemappings();
+
+    for (const from in remappings) {
+      if (Object.prototype.hasOwnProperty.call(remappings, from)) {
+        const to = remappings[from];
+        if (importName.startsWith(from)) {
+          return importName.replace(from, to);
+        }
+      }
+    }
+
+    return importName;
+  }
+);
+
+// Task that includes the remappings in solc input
+internalTask(TASK_COMPILE_SOLIDITY_GET_COMPILATION_JOB_FOR_FILE).setAction(
+  async (
+    {
+      dependencyGraph,
+      file,
+    }: {
+      dependencyGraph: DependencyGraph;
+      file: ResolvedFile;
+      solidityFilesCache?: SolidityFilesCache;
+    },
+    hre,
+    runSuper
+  ): Promise<CompilationJob | CompilationJobCreationError> => {
+    const job = (await runSuper({ dependencyGraph, file })) as
+      | CompilationJob
+      | CompilationJobCreationError;
+
+    if (!pluginActivated || isCompilationJobCreationError(job)) {
+      return job;
+    }
+
+    const remappings = getRemappings();
+    job.getSolcConfig().settings.remappings = Object.entries(remappings).map(
+      (fromTo) => fromTo.join("=")
+    );
+
+    return job;
+  }
+);
+
+task(
+  TASK_INIT_FOUNDRY,
+  "Initialize foundry setup in current hardhat project",
+  async (_, hre: HardhatRuntimeEnvironment) => {
+    if (existsSync("foundry.toml")) {
+      console.log(`File foundry.toml already exists. Aborting.`);
+      return;
+    }
+
+    console.log(`Creating foundry.toml file...`);
+
+    writeFileSync(
+      "foundry.toml",
+      [
+        `[profile.default]`,
```

Comment:
I think this would look better if we use a multi-line string and interpolation? Something like:

```
writeFileSync(
  "foundry.toml",
  `
[profile.default]
src = ${foundrySrc}
out = 'out'
```

etc.

I'm not 100% sure though.

Reply:
Normally I don't use that approach because it breaks indentation, but it's just personal preference. Your call

Reply:
I don't have a strong preference, let's keep it as it is.